src/LCRmeter-Keithley_590/main.py

```diff
@@ -0,0 +1,335 @@
+
+    def set_frequency(self, frequency: str) -> None:
+        """Set the measurement frequency."""
+        command = "F0X" if frequency == "100kHz" else "F1X"
+        self.port.write(command)
+
+    def set_range(self, measurement_range: str) -> None:
+        """Set the measurement range."""
+        if self.frequency == "100kHz":
+            # TODO: Add 10x amplifier boolean to GUI
+            ranges = {
+                "Auto": "R0",
+                "2pF/2uS": "R1",
+                "20pF/20uS": "R2",
+                "200pF/200uS": "R3",
+                "2nF": "R4",
+                "R1 x 10": "R5",
+                "R2 x 10": "R6",
+                "R3 x 10": "R7",
+                "R4 x 10": "R8",
+                "Auto off": "R9",
+            }
+        else:
+            ranges = {
+                "Auto": "R0",
+                "20pF/200uS": "R1",
+                # "20pF/200uS": "R2",  # doubled?
+                "200pF/2mS": "R3",
+                "2nF/20mS": "R4",
+                "Auto off": "R9",
+            }
+
+        try:
+            range_command = ranges[measurement_range]
+        except KeyError as e:
+            msg = f"Range {measurement_range} not available for frequency {self.frequency}."
+            range_command = ranges["Auto"]
+            raise ValueError(msg) from e
+
+        self.port.write(f"{range_command}X")
+
+    def set_reading_rate(self, rate: str) -> None:
+        """Set the reading rate."""
+        command = f"S{self.reading_rates[rate]}X"
+        self.port.write(command)
+
+    def set_trigger_mode_and_source(self, trigger: str) -> None:
+        """Set the trigger mode and source."""
+        command = f"T{self.trigger_modes[trigger]}X"
+        self.port.write(command)
+
+    def set_zero_on(self, set_on: bool = True) -> None:
+        """Set the zero."""
+        command = "Z1X" if set_on else "Z0X"
+        self.port.write(command)
+
+    def set_filter_on(self, filter_on: bool = True) -> None:
+        """Set the filter."""
+        command = "P1X" if filter_on else "P0X"
+        self.port.write(command)
+
+    def set_voltage(self, voltage: float, averaging: int = 1) -> None:
+        """Set the voltage.
+
+        Note: minimum step size to be applied is 5mV.
+        TODO: Add list mode, check which parameters are used for which waveform type (manual 3.14)
+        TODO: Verify if averaging is applied correctly in non-sweep mode.
+        """
+        max_voltage = 20
+        if abs(voltage) > max_voltage:
+            msg = f"Voltage {voltage} is out of range. Choose between -20 V and 20 V."
+            raise ValueError(msg)
+
+        # single voltage
+        command = f"V,,,{voltage},{averaging}X"  # Commas needed to skip start, stop, and step values
+        self.port.write(command)
+
+    def set_waveform(self) -> None:
+        """0 DC, 1 Single Staircase, 2 Dual Staircase, 3 Pulse Train, 4 External bias source."""
+        waveform = 1
```

Comment:
Should this function still get an argument? Otherwise the docstring does not really describe how the function is used, but it is rather an internal comment